ForumProject/projects/urls.py

```diff
@@ -8,5 +8,25 @@
 router.register('', views.ProjectViewSet)
 
 urlpatterns = [
-    path('', include(router.urls))
+    path('', include(router.urls)), 
+    path('files-of-project/<int:project>/', views.ProjectFilesViewSet.as_view({
+        'get': 'list',
+        'post': 'create',
+        'delete': 'destroy',
+    }), name='project_files_by_project'),
+    path('delete-file/<int:projectfiles_id>/of-project/<int:project>/',
+        views.delete_project_file, name='delete_project_file'),
+    path('shortlist-project/<int:project_id>/by-investor/<int:investor_id>/',
+        views.shortlist_project, name='shortlist_project'),
+    path('subscribe-for-project/<int:project_id>/by-investor/<int:investor_id>/for-share/<int:share>/',
+        views.subscribe_for_project, name='subscribe_for_project'),
+    path('delist-project/<int:project_id>/by-investor/<int:investor_id>/',
+        views.delist_project, name='delist_project'),
+    path('shortlisted-projects-of-startup/<int:startup_id>/', 
+        views.shortlisted_projects_of_startup, name='shortlisted_projects_of_startup'),
+    path('shortlisted-projects-of-investor/<int:investor_id>/',
+        views.shortlisted_projects_of_investor, name='shortlisted_projects_of_investor'),
+    path('logs/<int:project>/', views.ProjectFilesViewSet.as_view({
+        'get': 'list',
+    }), name='project_logs')
 ]
```

Comment:
Although it's more of a style choice, RESTful API design typically avoids using verbs in URLs (like delete, subscribe, delist) and instead relies on HTTP methods to indicate action. Consider if these can be refactored to use resource-based paths with appropriate HTTP methods. 
_Example_
```
urlpatterns = [
    path('', include(router.urls)),  
    path('projects/<int:project_id>/files/', views.ProjectFilesViewSet.as_view({
        'get': 'list',  # Get all files for a project
        'post': 'create',  # Add a new file to the project
    }), name='project_files'),

    path('investors/<int:investor_id>/projects/', views.InvestorProjectsViewSet.as_view({
        'get': 'list',  # Get all projects associated with an investor
    }), name='investor_projects'),

    path('investors/<int:investor_id>/projects/<int:project_id>/', views.InvestorProjectDetailViewSet.as_view({
        'put': 'update',  # Update details of a project for an investor
        'delete': 'destroy',  # Remove a project from an investor's list
    }), name='investor_project_detail'),

    # For subscribing or changing subscription details
    path('investors/<int:investor_id>/projects/<int:project_id>/subscribe/', views.SubscribeProjectViewSet.as_view({
        'post': 'create',  # Subscribe to a project
        'patch': 'partial_update',  # Change subscription details
        'delete': 'destroy',  # End subscription
    }), name='subscribe_project'),

    # View for accessing and managing project logs
    path('projects/<int:project_id>/logs/', views.ProjectLogsViewSet.as_view({
        'get': 'list',  # List logs for a project
    }), name='project_logs'),
]

```

Reply:
Renamings have been made